### ISSUE TITLE
json rewrite may not need slow t.co lookup

### ISSUE THREAD

**Issue by brad2014** [DocNow/twitter-archive-unshorten#1]
Thanks, @edsu, I had fun trying this out, and I noticed that the long URLs are apparentlly in the archive - I think there may be no need to do an expensive t.co lookup/decode.  It might suffice to read the json, and write the full_text json interpolating the urls json data into the full_text.  (or read the json to build the url_map in one pass, then do the substitution as you do now):

A quick URL map of tweets:

```
% sed -e '1s,.*= ,,' tweets.js | jq '.[].tweet.entities.urls[]? | {url, expanded_url}'
{
  "url": "https://t.co/g6ex748fSq",
  "expanded_url": "https://www.youtube.com/watch?v=orWsSWyiHAA"
}
```

I also notice that images point back to the twitter host, and may be worth rewriting to point to the local copy:

```
% sed -e '1s,.*= ,,' tweets.js | jq '.[].tweet.entities.media[]? | { url, expanded_url, media_url }'
{
  "url": "https://t.co/8IQRpVF7gN",
  "expanded_url": "https://twitter.com/a_drewsky/status/1583082040932474880/photo/1",
  "media_url": "http://pbs.twimg.com/media/Ffg8YSLUoAIFNNG.png"
}
...
```

One would rewrite `https://t.co/8IQRpVF7gN` to `./tweets_media/1583082040932474880/Ffg8YSLUoAIFNNG.png`.

I'll keep playing with it.

**Comment by edsu** [comment]
Thanks @brad2014 -- Yes, many of the short/long mappings are in the `data/tweet.js` file, but not all of them unfortunately. I'm about to release a new version that will use those when they are available which should hopefully speed it up a bit.

**Comment by edsu** [comment]
@brad2014 I like the idea of rewriting the image files: do you think `./tweets_media/1583082040932474880/Ffg8YSLUoAIFNNG.png` will work ok in the app?

**Comment by edsu** [comment]
I just released v0.0.7 that will use existing mappings from `tweet.js` instead of looking those up. It handles both `urls` and `media` in entities. I didn't have it rewrite images to the local file path till we can confirm that there is a way for it to work in the app. If you see a way forward with that @brad2014 please open another issue!
